tools/extract_baserom.py

```diff
@@ -1,1618 +1,1633 @@
+    "anime_texture_5_static",
+    "anime_texture_6_static",
+    "softsprite_matrix_static",
 ]
 
+
+romData = None
+
+
+def initialize_worker(rom_data):
+    global romData
+    romData = rom_data
+
+
 def read_uint32_be(offset):
-    return struct.unpack('>I', romData[offset:offset+4])[0]
+    return struct.unpack(">I", romData[offset : offset + 4])[0]
+
 
 def write_output_file(name, offset, size):
     try:
-        with open(name, 'wb') as f:
-            f.write(romData[offset:offset+size])
+        with open(name, "wb") as f:
+            f.write(romData[offset : offset + size])
     except IOError:
-        print('failed to write file ' + name)
-
-try:
-    os.mkdir('baserom')
-except:
-    pass
+        print("failed to write file " + name)
 
-# read baserom data
-try:
-    with open(ROM_FILE_NAME, 'rb') as f:
-        romData = f.read()
-except IOError:
-    print('failed to read file' + ROM_FILE_NAME)
-    sys.exit(1)
 
-# extract files
-for i in range(0, len(FILE_NAMES)):
-    filename = 'baserom/' + FILE_NAMES[i]
+def ExtractFunc(i):
+    filename = SEGMENTS_PATH / FILE_NAMES[i]
     entryOffset = FILE_TABLE_OFFSET + 16 * i
 
     virtStart = read_uint32_be(entryOffset + 0)
-    virtEnd   = read_uint32_be(entryOffset + 4)
+    virtEnd = read_uint32_be(entryOffset + 4)
     physStart = read_uint32_be(entryOffset + 8)
-    physEnd   = read_uint32_be(entryOffset + 12)
+    physEnd = read_uint32_be(entryOffset + 12)
 
-    if physStart == 0xFFFFFFFF and physEnd == 0xFFFFFFFF: # file deleted
+    if physStart == 0xFFFFFFFF:  # deleted
         if (virtEnd - virtStart) == 0:
-            continue
+            return
         physStart = virtStart
-        physEnd = 0
-        compressed = False
         size = virtEnd - virtStart
-    if physEnd == 0: # uncompressed
-        compressed = False
+    elif physEnd == 0:  # uncompressed
         size = virtEnd - virtStart
-    else: # compressed
-        compressed = True
+    else:  # compressed
         size = physEnd - physStart
 
-    # print(f"    0x{virtStart:08X}:(\"_{filename.split('/')[-1]}SegmentRomStart\",\"UNK_TYPE\",\"\",0x4),")
-    # print(f"    0x{virtEnd:08X}:(\"_{filename.split('/')[-1]}SegmentRomEnd\",\"UNK_TYPE\",\"\",0x4),")
-
-    #print(f"_{FILE_NAMES[i]}SegmentRomStart = {hex(virtStart)}\n_{FILE_NAMES[i]}SegmentRomEnd = {hex(virtEnd)}")
-    # print('extracting ' + filename + " (0x%08X, 0x%08X)" % (virtStart, virtEnd))
+    print(f"extracting {filename} (0x{virtStart:08X}, 0x{virtEnd:08X})")
     write_output_file(filename, physStart, size)
-    if compressed:
-        os.system('tools/buildtools/yaz0 -d ' + filename + ' ' + filename)
+
+
+#####################################################################
+
+
+def main():
+    SEGMENTS_PATH.mkdir(parents=True, exist_ok=True)
+
+    # read baserom data
+    try:
+        rom_data = ROM_FILE_PATH.read_bytes()
+    except IOError:
+        print(f"failed to read file {ROM_FILE_PATH}")
+        exit(1)
```

Comment:
I think it would be better if we checked if the file exists instead with `.exists()`